exporters/jaeger/src/recordable.cc

```diff
@@ -39,12 +39,25 @@ void Recordable::PopulateAttribute(nostd::string_view key, const common::Attribu
 void Recordable::SetIdentity(const trace::SpanContext &span_context,
                              trace::SpanId parent_span_id) noexcept
 {
+  // IDs should be converted to big endian before transmission.
+  // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk_exporters/jaeger.md#ids
+#if JAEGER_IS_LITTLE_ENDIAN == 1
```

Comment:
I'm wondering if this is something that should be handled by `htons`?

https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-htons
https://linux.die.net/man/3/htons

Reply:
Or, to be more precise, something like this?
```cpp
#define htonll(x) ((1==htonl(1)) ? (x) : ((uint64_t)htonl((x) & 0xFFFFFFFF) << 32) | htonl((x) >> 32))
#define ntohll(x) ((1==ntohl(1)) ? (x) : ((uint64_t)ntohl((x) & 0xFFFFFFFF) << 32) | ntohl((x) >> 32))
```

Reply:
nicely done. this will check the endianness to before performing conversion

Reply:
Thanks. I considered `htonl` as the first option, but I'd like to avoid the runtime check on endianness and preferred the compiler intrinsic to perform byte order swap for simplicity and slight perf win.

Reply:
I could change the function name to `to_big_endian` and define it as as direct return for `big endian` host.  In this way, we don't need check the endianness at callsite of this function.

Reply:
Second thought, leave the `JAEGER_IS_LITTLE_ENDIAN` in the callsite could save us to create `bswap_128`.

Reply:
I was wondering if it's something you can move into a separate static inline utility function, then it can be reused elsewhere? I think it might be common requirement, to transmit the buffer in big endian, not unique to Jaeger?

Reply:
So far I only saw the requirement in Jaeger. I'd be happen to move this to common utility if it is used in other places.